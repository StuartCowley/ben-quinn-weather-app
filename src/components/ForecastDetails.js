import React from "react";
import PropTypes from "prop-types";
import "../styles/ForecastDetails.css";

function ForecastDetails({ forecasts }) {
  const { date, temperature, humidity, wind } = forecasts;
  const formattedDate = new Date(date).toDateString();

  return (
    <div className="forecast-details">
      <div className="forecast-details__date">{formattedDate}</div>
      <div className="forecast-details__max-temperature">
        <b>Max Temperature:</b> {temperature.max}&deg;C
      </div>
      <div className="forecast-details__min-temperature">
        <b>Min Temperature:</b> {temperature.min}&deg;C
      </div>
      <div className="forecast-details__humidity">
        <b>Humidity:</b> {humidity}%
      </div>
      <div className="forecast-details__wind">
        <b>Wind:</b> {wind.speed}mph {wind.direction}
      </div>
    </div>
  );
}

export default ForecastDetails;

ForecastDetails.propTypes = {
  forecasts: PropTypes.shape({
    date: PropTypes.number,
    temperature: PropTypes.shape({
      min: PropTypes.number,
      max: PropTypes.number,
    }),
    humidity: PropTypes.number,
    wind: PropTypes.shape({
      speed: PropTypes.number,
      direction: PropTypes.string,
    }),
  }).isRequired,
};
