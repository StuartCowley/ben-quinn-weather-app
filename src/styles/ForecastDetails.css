.forecast-details {
  display: flex;
  align-items: center;
  flex-direction: column;
}

.forecast-details__date {
  font-size: 2em;
  font-weight: bold;
}
