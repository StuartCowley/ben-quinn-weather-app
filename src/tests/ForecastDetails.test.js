import React from "react";
import { render } from "@testing-library/react";
import ForecastDetails from "../components/ForecastDetails";

describe("ForecastDetails", () => {
  const validProps = {
    date: 1111111,
    temperature: {
      min: 12,
      max: 22,
    },
    humidity: 90,
    wind: {
      speed: 10,
      direction: "s",
    },
  };

  it("renders correctly", () => {
    const { asFragment } = render(<ForecastDetails forecasts={validProps} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct values for props", () => {
    const { getByText } = render(<ForecastDetails forecasts={validProps} />);

    expect(getByText(/January 1st 1970/i)).toHaveClass(
      "forecast-details__date"
    );
    expect(getByText(/12°C/)).toHaveClass("forecast-details__min-temperature");
    expect(getByText(/22°C/)).toHaveClass("forecast-details__max-temperature");
    expect(getByText(/90%/)).toHaveClass("forecast-details__humidity");
    expect(getByText(/10mph/)).toHaveClass("forecast-details__wind");
  });
});
