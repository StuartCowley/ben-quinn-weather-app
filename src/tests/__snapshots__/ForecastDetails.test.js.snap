// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastDetails renders correctly 1`] = `
<DocumentFragment>
  <div
    class="forecast-details"
  >
    <div
      class="forecast-details__date"
    >
      January 1st 1970
    </div>
    <div
      class="forecast-details__max-temperature"
    >
      <b>
        Max Temperature:
      </b>
       22°C
    </div>
    <div
      class="forecast-details__min-temperature"
    >
      <b>
        Min Temperature:
      </b>
       12°C
    </div>
    <div
      class="forecast-details__humidity"
    >
      <b>
        Humidity:
      </b>
       90%
    </div>
    <div
      class="forecast-details__wind"
    >
      <b>
        Wind:
      </b>
       10mph s
    </div>
  </div>
</DocumentFragment>
`;
